src/main.rs

```diff
@@ -0,0 +1,188 @@
+            ))),
+            _ => Err(err),
+        }
+    }
+}
+
+fn main() {
+    // Lazy static, make sure they don't panic
+    let _ = &*BIRTHDAY;
+    let _ = &*IMAGE_JSON;
+
+    let s = Static::new(Path::new("./static/"));
+
+    let mut router = Router::new();
+    router.any("/code.tar.gz", source_code, "source code");
+    router.any("/index.php", index, "backwards-compat-index");
+    router.any("/", index, "index");
+    router.any("/*", s, "static");
+    router.get("/healthz", healthz, "healthz");
+
+    let mut chain = iron::middleware::Chain::new(router);
+    chain.link_after(RMS404Handler);
+
+    let _server = Iron::new(chain).http("0.0.0.0:3000").unwrap();
+    println!("Listening on 3000");
+}
+
+// AGPL compliance. We exclude the images to avoid increasing build and download size too much;
+// people can easily enough download them from the repo or via checking the image json regardless.
+include_repo_gz!(SOURCE_CODE, ".", ":!**/*.jpg", ":!**/*.JPG", ":!**/*.png");
+
+fn source_code(_: &mut Request) -> IronResult<Response> {
+    Ok(Response::with((
+        headers::ContentType(mime::Mime(
+            mime::TopLevel::Application,
+            mime::SubLevel::Ext("gzip".to_string()),
+            vec![],
+        )).0,
+        status::Ok,
+        &SOURCE_CODE[..],
+    )))
+}
+
+fn healthz(_: &mut Request) -> IronResult<Response> {
+    Ok(Response::with((status::Ok, "healthy")))
+}
+
+fn index(req: &mut Request) -> IronResult<Response> {
+    let js_enabled = match req.get_ref::<UrlEncodedQuery>() {
+        Ok(query) => {
+            if query.contains_key("images") {
+                return list_images();
+            }
+            query.contains_key("js")
+        }
+        Err(e) => {
+            warn!("Could not decode query: {}", e);
+            false
+        }
+    };
+    let now = Utc::now();
+    let title = if BIRTHDAY.month() == now.month() && BIRTHDAY.day() == now.day() {
+        "Happy GNU/Birthday"
+    } else {
+        "Our GNU/Lord and GNU/Savior is 100% sexy!"
+    };
+
+    let resp = format!(
+        r#"<!DOCTYPE html>
+<!-- SPDX-License-Identifier: AGPL-3.0-only -->
+<html>
+	<head>
+		<!--
+This program is free software: you can redistribute it and/or  modify
+it under the terms of the GNU Affero General Public License, version 3,
+as published by the Free Software Foundation.
+The source code, along with the full license text, is available here:
+https://github.com/euank/rms.sexy
+
+It is also available at "/code.tar.gz" on this server.
+		-->
+		<meta charset="utf-8">
+		<title>{}</title>
+		{}
+		<link rel="stylesheet" href="/style.css">
+		<link rel="license" href="/license.txt">
+		<script async src="/script.js"></script>
+	</head>
+	<body>
+		<a href="https://donate.fsf.org/"><img class="donate" src="/donate.png" alt="Donate!" title="Donate to the FSF!"></a>
+		<a href="/"><img alt="RMS Matthew Stallman" class="stallman" src="{}" height="100%"></a>
+	</body>
+</html>"#,
```

Comment:
Why not store the HTML in a separate file?

Reply:
There's not really a need to since it's so short.

Even if I stored it as a file, I'd want to `include!` it into the built binary to be able to just serve it from memory without any disk IO.
I don't think it matters much either way. Is there a reason you think it's better stored as a file?